channel/erc20_depositor.go

```diff
@@ -40,42 +45,138 @@ const ERC20DepositorTXGasLimit = 100000
 // Return value of ERC20Depositor.NumTx.
 const erc20DepositorNumTx = 2
 
+// Keep track of the increase allowance and deposit processes.
+var mu sync.Mutex
+var locks = make(map[string]*sync.Mutex)
```

Comment:
Please give them more precise namings. As far as I understand it from the code now is that the first mutex is only the mutex for the "locks" map?

```suggestion
var depositLocksMtx sync.Mutex
var depositLocks = make(map[string]*sync.Mutex)
```

Reply:
Or even better, create a custom struct (which is then initialized in `NewERC20Depositor`).
```suggestion
// TODO: Add a descriptive doc here explaining the purpose and reason of this lock.
type DepositLocks struct {
	mtx sync.Mutex
	locks map[string]*sync.Mutex
}
```